src/layerInfos/TarifverbundkarteTopicInfo/TarifVerbundLegend.js

```diff
@@ -0,0 +1,68 @@
+import React, { useEffect, useState, useMemo } from 'react';
+import { useSelector } from 'react-redux';
+import { makeStyles } from '@material-ui/core';
+
+import TarifverbundPartner from '../../components/TarifverbundPartner';
+
+const useStyles = makeStyles(() => {
+  return {
+    legend: {
+      display: 'grid',
+      gridTemplateColumns: (props) =>
+        props.isMobile ? '1fr 1fr' : '200px 200px',
+      gap: 5,
+    },
+  };
+});
+
+const useFetchVerbunde = (vectorTilesUrl) => {
+  const [verbunde, setVerbunde] = useState([]);
+  useEffect(() => {
+    fetch(`${vectorTilesUrl}/styles/ch.sbb.tarifverbund/style.json`)
+      .then((response) => response.json())
+      .then((data) =>
+        setVerbunde(
+          data.metadata.partners.sort((a, b) => a.name.localeCompare(b.name)),
+        ),
+      )
+      // eslint-disable-next-line no-console
+      .catch((err) => console.error(err));
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
can you cancel the request when necessary  ?

Reply:
maybe return an empty array when the request fails

Reply:
The defaultIt already returns an empty array, since the default state in useState is set to an empty array